src/compression.jl

```diff
@@ -96,12 +86,22 @@ function TranscodingStreams.finalize(codec::ZstdCompressor)
         end
         codec.cstream.ptr = C_NULL
     end
-    reset!(codec.cstream.ibuffer)
-    reset!(codec.cstream.obuffer)
-    return
+    nothing
 end
 
 function TranscodingStreams.startproc(codec::ZstdCompressor, mode::Symbol, error::Error)
+    if codec.cstream.ptr == C_NULL
+        ptr = LibZstd.ZSTD_createCStream()
+        if ptr == C_NULL
+            throw(OutOfMemoryError())
+        end
+        codec.cstream.ptr = ptr
+        i_code = initialize!(codec.cstream, codec.level)
```

Comment:
Also see notes in #73

Should `initialize!` throw so we can catch it here and transmit the error?